src/api/order/validations/findOrder.js

```diff
@@ -13,6 +14,8 @@ async function findOrder(req, res, next) {
     const order = await Order.findOne({ _id: id });
     if (!order) {
       return sendFailedResponse(res);
+    } else if (USER_ROLES.user === user.role && order.userId.toString() !== user._id.toString()) {
+      return sendFailedResponse(res, HTTP_STATUSES.FORBIDDEN.message, HTTP_STATUSES.FORBIDDEN.code);
     }
```

Comment:
Why don't you just put userId in filter, when user.role is user? Why are you checking like this?You just need to check, if role is USER, add suerId in filter, so it will not find the order, if userIds are different